app/scripts/services/resourceAlerts.js

```diff
@@ -100,6 +101,33 @@ angular.module("openshiftConsole")
       }
     };
 
+    // deploymentConfig, k8s deployment
+    var getPausedDeploymentAlerts = function(alerts, deployment) {
```

Comment:
Agree!  Second one is modifying the array:

```javascript
    notifications = ResourceAlertsService.getDeploymentStatusAlerts(deploymentConfig, mostRecent);
    notifications = ResourceAlertsService.getPausedDeploymentAlerts(notifications, deploymentConfig);
```
Maybe it would be better like this:
```javascript
var notifications = {};
angular.extend(notifications, ResourceAlertsService.getDeploymentStatusAlerts(deploymentConfig, mostRecent), ResourceAlertsService.getPausedDeploymentAlerts(deploymentConfig))
```
And eliminate the need to pass in the alerts inconsistently.